test/integration_test/px_statfs_webhook_test.go

```diff
@@ -153,3 +167,106 @@ func runCommandInSv4TestContainer(t *testing.T, pod *corev1.Pod, cmd []string) s
 func isVirtLauncherContext(ctx *scheduler.Context) bool {
 	return strings.HasPrefix(ctx.App.Key, "virt-launcher-sim")
 }
+
+// Make dummy changes to the px-statfs configmap and then delete a virt-launcher-sim pod.
+// Stork should restore the original configMap.
+func validateConfigMapUpdate(t *testing.T, ctx *scheduler.Context) {
+	require.True(t, isVirtLauncherContext(ctx))
+	vols, err := schedulerDriver.GetVolumes(ctx)
+	require.NoError(t, err, "Failed to get volumes for context %s", ctx.App.Key)
+
+	require.Equal(t, len(vols), 1)
+	vol := vols[0]
+
+	updateFn1 := func(cm *corev1.ConfigMap) {
+		// change the contents of ld.so.preload file in the configmap
+		cm.Data[cmDataKey] = "dummy-value"
+	}
+	updateFn2 := func(cm *corev1.ConfigMap) {
+		// change the contents of px_statfs.so in the configMap
+		cm.BinaryData[cmBinaryDataKey] = []byte("dummy-value")
+	}
+	updateFn3 := func(cm *corev1.ConfigMap) {
+		// change value of a single byte in px_statfs.so
+		origBytes := cm.BinaryData[cmBinaryDataKey]
+		updatedBytes := make([]byte, len(origBytes))
+		for i := range origBytes {
+			updatedBytes[i] = origBytes[i]
+		}
+		updateIndex := len(updatedBytes) / 2
+		origByte := updatedBytes[updateIndex]
+		newByte := origByte ^ 1
+		logrus.Infof("Updating byte at index %d from %d to %d", updateIndex, origByte, newByte)
+		updatedBytes[updateIndex] = newByte
+		cm.BinaryData[cmBinaryDataKey] = updatedBytes
+	}
+	for i, updateFn := range []func(*corev1.ConfigMap){updateFn1, updateFn2, updateFn3} {
+		logrus.Infof("Testing configmap update function %d", i)
+		pods, err := core.Instance().GetPodsUsingPV(vol.ID)
+		require.NoError(t, err, "Failed to get pods for volume %v of context %s", vol.ID, ctx.App.Key)
+
+		require.True(t, len(pods) > 0)
+
+		// pick a pod that is not terminating already
+		var pod corev1.Pod
+		for _, pod = range pods {
+			if pod.DeletionTimestamp.IsZero() {
+				logrus.Infof("Using pod %s/%s for the configMap update test", pod.Namespace, pod.Name)
+				break
+			}
+		}
+
+		logrus.Infof("Changing configMap in namespace %s", pod.Namespace)
+		cm, err := core.Instance().GetConfigMap(statfsConfigMapName, pod.Namespace)
+		require.NoError(t, err)
+
+		origData, exists := cm.Data[cmDataKey]
+		require.True(t, exists)
+
+		origBinaryData, exists := cm.BinaryData[cmBinaryDataKey]
+		require.True(t, exists)
+
+		// change value of either data key or the binaryData and verify that it gets restored to the original value
+		// when a new pod starts up
+		updateFn(cm)
+		cm, err = core.Instance().UpdateConfigMap(cm)
+		require.NoError(t, err)
+
+		err = core.Instance().DeletePod(pod.Name, pod.Namespace, false)
+		require.NoError(t, err)
+
+		require.Eventuallyf(t, func() bool {
+			return configMapMatches(pod.Namespace, origData, origBinaryData)
+		}, 3*time.Minute, 10*time.Second, "ConfigMap %s/%s did not get updated", pod.Namespace, statfsConfigMapName)
```

Comment:
Help me understand this. From what I understand, origData and origBinaryData is referring to older data in the cm.  updateFn should update it and we should have updatedData and updatedBinaryData against which we should compare in  configMapMatches. What am I missing?

Reply:
Adding the following comment to that function to clarify this. Let me know if this makes sense:
```
// This test verifies that the webhook updates the px-statfs configMap, if its contents do not match
// the expected values.
// Test steps:
//
// - Start with a "good" px-statfs configmap that the webhook has already created.
// - Make some dummy changes to the px-statfs configmap
// - Trigger the webhook by deleting one of the virt-launcher-sim pod (new pod should start)
// - Wait until the webhook has restored the original configMap
```

Reply:
Makes sense.